syntax/javascript.vim

```diff
@@ -214,14 +214,18 @@ if main_syntax == "javascript"
 endif
 
 exe 'syntax match jsFunction /\<function\>/ nextgroup=jsGenerator,jsFuncName,jsFuncArgs skipwhite '.(exists('g:javascript_conceal_function') ? 'conceal cchar='.g:javascript_conceal_function : '')
+exe 'syntax match jsArrowFunction /=>/ skipwhite nextgroup=jsFuncBlock contains=jsFuncBraces '.(exists('g:javascript_conceal_arrow_function') ? 'conceal cchar='.g:javascript_conceal_arrow_function : '')
 
-syntax match   jsGenerator      contained '\*' nextgroup=jsFuncName skipwhite
+syntax match   jsGenerator      contained '\*' nextgroup=jsFuncName,jsFuncArgs skipwhite
 syntax match   jsFuncName       contained /\<[a-zA-Z_$][0-9a-zA-Z_$]*/ nextgroup=jsFuncArgs skipwhite
 syntax region  jsFuncArgs       contained matchgroup=jsFuncParens start='(' end=')' contains=jsFuncArgCommas,jsFuncArgRest,jsAssignmentExpr,jsComment,jsLineComment nextgroup=jsFuncBlock keepend skipwhite skipempty
 syntax match   jsFuncArgCommas  contained ','
 syntax match   jsFuncArgRest    contained /\%(\.\.\.[a-zA-Z_$][0-9a-zA-Z_$]*\))/
 
-exe 'syntax match jsArrowFunction /=>/ '.(exists('g:javascript_conceal_arrow_function') ? 'conceal cchar='.g:javascript_conceal_arrow_function : '')
+" Matches a single keyword argument with no parens
+syntax match   jsArrowFuncArgs  /\(\k\)\+\s\+\(=>\)\@=/ skipwhite contains=jsFuncArgs extend
+" Matches a series of arguments surrounded in parens
+syntax match   jsArrowFuncArgs  /(\(\k\|,\|\s\|\n\|\.\)*)\s\+\(=>\)\@=/ skipwhite contains=jsFuncArgs extend
```

Comment:
This is used to match the arguments definition for an `=>` function.

Currently for normal functions we actually highlight the parenthesis and actual arguments definitions, see this example:

![func arg example](https://dl.dropboxusercontent.com/s/hke7zyv7iu58bar/file.js_2016-04-03_22-01-48.png?dl=0)

My color theme makes them orange.

To match this for lambdas, it's a bit trickier. We have to use a lookahead to match against the `=>`.

I separated it into 2 lines to make the overall regexes easier to understand.

Lambda functions can be both defined like this:

```js
// The first of the syntax match lines matches this case
arg => obj.getter();
```

And like this:

```js
// The second syntax match line matches for this
(arg) => obj.getter()
```